trio/_core/_mock_clock.py

```diff
@@ -62,7 +62,7 @@ class MockClock(Clock, metaclass=Final):
 
     """
 
-    def __init__(self, rate: float = 0.0, autojump_threshold: float = inf):
+    def __init__(self, rate: float = 0.0, autojump_threshold: float = inf) -> None:
```

Comment:
Err, are we doing this or not...? Maybe put this in your bigger PR about these and punt decisions to that central place?

Reply:
With the changes I made with `verify_types.json`, it said that this PR adds 2 new typed functions, so I think this change will be important for 100% type completion for the entire project

Reply:
Oh does pyright require this? Could you check please? If so, I'm totally on board with adding `-> None` everywhere: that adds software to ensure we are consistently putting the `-> None` everywhere.

Reply:
This section of pep 484 talks about `__init__ -> None` annotations: https://peps.python.org/pep-0484/#the-meaning-of-annotations

Reply:
> With the changes I made with `verify_types.json`, it said that this PR adds 2 new typed functions, so I think this change will be important for 100% type completion for the entire project

That's because it counts the class itself as another symbol that's now fully typed. If you reverted only this one change it wouldn't change the output.

`mypy` has resolved that parenthetical in a different way, and seems like `pyright` does too here, in them not requiring `-> None` if all other parameters to `__init__` are typed. I can't even get `mypy` to complain if I add `--strict`, so I'm a bit curious what configuration you're using that flags it.